controllers/user.server.controller.js

```diff
@@ -36,10 +36,48 @@ exports.userLogout = function(req, res) {
  * @param res HTTP request response object
  */
 exports.userViewById = function(req, res) {
-    // TODO: implement userViewById()
-    res.json({ dummyTest: "userViewById() dummy test passes" });
+    const { id } = req.params;
+
+    User.findById(id)
+        .then(user => {
+            if (user) {
+                res.json(user);
+            }
+            else {
+                res.sendStatus(404);
+            }
+        })
+        .catch(err => res.sendStatus(400).json('Error: ' + err));
 }
```

Comment:
This function fails an existing test. This does not mean the function fails, because it only tested for a hardcoded string to be returned. However, I think the test should be updated to test for what is expected to be returned to the front end by the changes made to the function.